src/monodroid/jni/pinvoke-override-api.cc

```diff
@@ -0,0 +1,709 @@
+	}
+	ret->period.mark_start ();
+
+	return ret;
+}
+
+static void
+monodroid_timing_stop (managed_timing_sequence *sequence, const char *message)
+{
+	static constexpr const char DEFAULT_MESSAGE[] = "Managed Timing";
+
+	if (sequence == nullptr)
+		return;
+
+	sequence->period.mark_end ();
+	Timing::info (sequence->period, message == nullptr ? DEFAULT_MESSAGE : message);
+	timing->release_sequence (sequence);
+}
+
+static char**
+monodroid_strsplit (const char *str, const char *delimiter, size_t max_tokens)
+{
+	return utils.monodroid_strsplit (str, delimiter, max_tokens);
+}
+
+static void
+monodroid_strfreev (char **str_array)
+{
+	utils.monodroid_strfreev (str_array);
+}
+
+static char*
+monodroid_strdup_printf (const char *format, ...)
+{
+	va_list args;
+
+	va_start (args, format);
+	char *ret = utils.monodroid_strdup_vprintf (format, args);
+	va_end (args);
+
+	return ret;
+}
+
+static char*
+monodroid_TypeManager_get_java_class_name (jclass klass)
+{
+	return monodroidRuntime.get_java_class_name_for_TypeManager (klass);
+}
+
+static void
+monodroid_store_package_name (const char *name)
+{
+	utils.monodroid_store_package_name (name);
+}
+
+static int
+monodroid_get_namespaced_system_property (const char *name, char **value)
+{
+	return static_cast<int>(androidSystem.monodroid_get_system_property (name, value));
+}
+
+static FILE*
+monodroid_fopen (const char* filename, const char* mode)
+{
+	return utils.monodroid_fopen (filename, mode);
+}
+
+static int
+send_uninterrupted (int fd, void *buf, int len)
+{
+	if (len < 0)
+		len = 0;
+	return utils.send_uninterrupted (fd, buf, static_cast<size_t>(len));
+}
+
+static int
+recv_uninterrupted (int fd, void *buf, int len)
+{
+	if (len < 0)
+		len = 0;
+	return static_cast<int>(utils.recv_uninterrupted (fd, buf, static_cast<size_t>(len)));
+}
+
+static void
+set_world_accessable (const char *path)
+{
+	utils.set_world_accessable (path);
+}
+
+static void
+create_public_directory (const char *dir)
+{
+	utils.create_public_directory (dir);
+}
+
+static char*
+path_combine (const char *path1, const char *path2)
+{
+	return utils.path_combine (path1, path2);
+}
+
+static void*
+monodroid_dylib_mono_new ([[maybe_unused]] const char *libmono_path)
+{
+	return nullptr;
+}
+
+static void
+monodroid_dylib_mono_free ([[maybe_unused]] void *mono_imports)
+{
+	// no-op
+}
+
+/*
+  this function is used from JavaInterop and should be treated as public API
+  https://github.com/xamarin/java.interop/blob/master/src/java-interop/java-interop-gc-bridge-mono.c#L266
+
+  it should also accept libmono_path = nullptr parameter
+*/
+static int
+monodroid_dylib_mono_init (void *mono_imports, [[maybe_unused]] const char *libmono_path)
+{
+	if (mono_imports == nullptr)
+		return FALSE;
+	return TRUE;
+}
+
+static void*
+monodroid_get_dylib (void)
+{
+	return nullptr;
+}
+
+#define PINVOKE_SYMBOL(_sym_) { #_sym_, reinterpret_cast<void*>(&_sym_) }
+
+MonodroidRuntime::pinvoke_api_map MonodroidRuntime::xa_pinvoke_map = {
```

Comment:
Does this table need to be sorted?  If so, that should be commented.

Reply:
No, it doesn't need to be sorted. `robin-map` is an equivalent to `std::unordered_map`, I sorted it just for convenience.